statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+                    # If there is only one or two clusters then weights wont change
+                    if (self.design.ncs[s] == 1 or self.design.ncs[s] == 2):
+                        continue
+                    # array of clusters to resample from
+                    ii = np.flatnonzero(self.design.sfclust == s)
+                    # resample them
+                    ii_resample = np.random.choice(ii, size=(self.design.ncs[s]-1))
+                    # accumulate number of times cluster i was resampled
+                    bin += np.bincount(ii_resample,
+                                       minlength=max(self.design.sclust)+1)
+            # avg number of times cluster i was resampled
+            bin /= replicates
+            # augment weights
+            w *= bin[self.design.sclust]
+            # call the stat w/ the new weights
+            if index is None:
+                jdata.append(self._stat(weights=w))
+            else:
+                jdata.append(self._stat(w, index))
+        else:
+            for r in range(replicates):
+                w = self.design.weights.copy()
+                bin = np.zeros(self.design.nclust)
+                for s in range(self.design.nstrat):
+                    # how to handle strata w/ only one cluster?
+                    w[self.design.strat == s] *= float(self.design.ncs[s] - 1) \
+                                                 / self.design.ncs[s]
+                    # If there is only one or two clusters then weights wont change
+                    if (self.design.ncs[s] == 1 or self.design.ncs[s] == 2):
+                        continue
+                    # array of clusters to resample from
+                    ii = np.flatnonzero(self.design.sfclust == s)
+                    # resample them
+                    ii_resample = np.random.choice(ii, size=(self.design.ncs[s]-1))
+                    # accumulate number of times cluster i was resampled
+                    bin += np.bincount(ii_resample,
+                                       minlength=max(self.design.sclust)+1)
+                # augment weights
+                w *= bin[self.design.sclust]
+                # call the stat w/ the new weights
+                if index is None:
+                    jdata.append(self._stat(weights=w))
+                else:
+                    jdata.append(self._stat(w, index))
+
+                if index is None:
+                    jdata.append(self._stat(weights = w))
+                else:
+                    jdata.append(self._stat(w, index))
+
+        jdata = np.asarray(jdata)
+        if self.mse:
+            print("mse specified")
+            var = np.dot((jdata-est).T, (jdata-est)) / (replicates)
+            var = np.sqrt(np.diag(var))
+        else:
+            boot_mean = jdata.mean(0)
+            var = ((jdata - boot_mean)**2).sum(0) / (replicates)
+
+        return est, var
+
+    def jack(self, index=None):
```

Comment:
Let's go with `jackknife` instead of `jack` (we use `bootstrap` not `boot`, and the number of characters is the same).